xcdat/bounds.py

```diff
@@ -560,91 +573,86 @@ def get_daily_time_bounds(time, frequency=1):
-        for f in range(frequency):
-            if f * (24 // frequency) <= h < (f + 1) * (24 // frequency):
-                l_bnd = cf_obj(y, m, d, f * (24 // frequency))
-                u_bnd = l_bnd + datetime.timedelta(hours=(24 // frequency))
-        # store
+        for f in range(freq):
+            if f * (24 // freq) <= h < (f + 1) * (24 // freq):
+                l_bnd = cf_obj(y, m, d, f * (24 // freq))
+                u_bnd = l_bnd + datetime.timedelta(hours=(24 // freq))
         time_bnds.append([l_bnd, u_bnd])
 
-    # create dataarray
-    time_bnds = xr.DataArray(  # type: ignore
+    da_time_bnds = xr.DataArray(
         name=f"{time.name}_bnds",
         data=time_bnds,
         coords=dict(time=time),
         dims=[*time.dims, "bnds"],
         attrs={"xcdat_bounds": "True"},
     )
 
-    return time_bnds
+    return da_time_bnds
 
 
-def get_time_bounds(time):
-    """Sets the time bounds for a time coordinate axis.
+def create_time_bounds(time: xr.DataArray) -> xr.DataArray:
+    """Creates time bounds for each timestep of the time coordinate axis.
+
+    This function uses ``_infer_freq`` to determine the temporal resolution and
+    then calls ``get_yearly_time_bounds``, ``get_monthly_time_bounds``, or
+    ``get_daily_time_bounds`` as appropriate. See the docstrings for these
+    functions for more details.
 
     Parameters
     ----------
     time : xr.DataArray
         The temporal coordinate variable for the axis.
-    frequency : int, optional, default 1
-        Flag to note set the number of timepoints per day.
 
     Returns
     -------
     xr.DataArray
         The bounds for the time axis.
 
-    Notes
-    -----
-    Function uses `_infer_freq` to determine the temporal resolution
-    and the calls `get_yearly_time_bounds`, `get_monthly_time_bounds`,
-    or `get_daily_time_bounds` as appropriate. See these functions for
-    more details.
+    Examples
+    --------
+    # TODO
     """
     freq = _infer_freq(time)
     if freq == "year":
-        time_bnds = get_yearly_time_bounds(time)
-    if freq == "month":
-        time_bnds = get_monthly_time_bounds(time)
-    if freq == "day":
-        time_bnds = get_daily_time_bounds(time)
-    if freq == "hour":
-        time_bnds = get_daily_time_bounds(time)
-        # get number of time steps per day
+        time_bnds = create_yearly_time_bounds(time)
+    elif freq == "month":
+        time_bnds = create_monthly_time_bounds(time)
+    elif freq == "day":
+        time_bnds = create_daily_time_bounds(time)
+    elif freq == "hour":
+        # Determine the daily frequency for generating time  bounds.
         diff = time.values[1] - time.values[0]
         hrs = diff.seconds / 3600
         daily_freq = int(24 / hrs)
-        # get sub-daily time bounds
-        time_bnds = get_daily_time_bounds(time, frequency=daily_freq)
+
+        time_bnds = create_daily_time_bounds(time, freq=daily_freq)  # type: ignore
 
```

Comment:
I forgot to mention that we only need to call `create_daily_time_bounds()` once here.

Reply:
You mean by combining "hour" and "day"? I kind of like the logical separation here.